### PR TITLE
Support Comdat sections on Windows

### PR DESCRIPTION
**Pull request by nattress**
Add a parameter to CreateCustomSection specifying a Comdat symbol to use
for the section being created. The selection policy is left as SelectAny
to allow the linker to pick.

### CODE REVIEW

**Comment by nattress** [comment]
@kyulee1 Please take a look

File: lib/ObjWriter/objwriter.cpp [hunk 1]
```diff
@@ -259,7 +259,8 @@ enum CustomSectionAttributes : int32_t {
 };
 
 extern "C" bool CreateCustomSection(ObjectWriter *OW, const char *SectionName,
-                                    CustomSectionAttributes attributes) {
+                                    CustomSectionAttributes attributes,
+                                    const char *ComdatName) {
```
Comment (kyulee1) [inline lib/ObjWriter/objwriter.cpp]: ComdateName is for Windows/Coff only.
Can you add an assert at minimum if ComdatName is not null like?:
```
assert(OW->MOFI->getObjectFileType() == OW->MOFI->IsCOFF);
```

Comment (nattress) [inline lib/ObjWriter/objwriter.cpp]: Done

**Comment by kyulee1** [comment]
LGTM. Test failures appear from lab script bugs.

**Pull request merged**